File: linkgame-common/src/main/java/io/github/octcarp/linkgame/common/packet/SimpStatus.java
package io.github.octcarp.linkgame.common.packet;

public enum SimpStatus {
    OK(200, "Request succeeded"),
    FAILURE(500, "Internal server error"),
    TIMEOUT(408, "Request timeout"),
    UNAUTHORIZED(401, "Unauthorized"),
    FORBIDDEN(403, "Forbidden"),
    NOT_FOUND(404, "Not found"),
    CONFLICT(409, "Conflict"),
    LOCKED(423, "Resource Locked"),
    ;

    private int code;
    private String message;

    SimpStatus(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public int getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }
}

File: game-common/src/main/kotlin/io/github/octcarp/sustech/cs209a/linkgame/common/packet/SimpStatus.kt
package io.github.octcarp.sustech.cs209a.linkgame.common.packet

import java.io.Serializable

enum class SimpStatus(
    val code: Int,
    val message: String
) : Serializable {
    OK(200, "Request succeeded"),
    FAILURE(500, "Internal server error"),
    TIMEOUT(408, "Request timeout"),
    UNAUTHORIZED(401, "Unauthorized"),
    FORBIDDEN(403, "Forbidden"),
    NOT_FOUND(404, "Not found"),
    CONFLICT(409, "Conflict"),
    LOCKED(423, "Resource Locked")
}
